examples/src/main/java/org/apache/spark/examples/mllib/JavaStreamingTestExample.java

```diff
@@ -94,22 +97,21 @@ public BinarySample call(String line) throws Exception {
     // $example off$
 
     // Stop processing if test becomes significant or we time out
-    final Accumulator<Integer> timeoutCounter =
-      ssc.sparkContext().accumulator(numBatchesTimeout);
+    timeoutCounter = numBatchesTimeout;
 
     out.foreachRDD(new VoidFunction<JavaRDD<StreamingTestResult>>() {
       @Override
-      public void call(JavaRDD<StreamingTestResult> rdd) throws Exception {
-        timeoutCounter.add(-1);
+      public void call(JavaRDD<StreamingTestResult> rdd) {
+        timeoutCounter -= 1;
 
-        long cntSignificant = rdd.filter(new Function<StreamingTestResult, Boolean>() {
+        boolean anySignificant = ! rdd.filter(new Function<StreamingTestResult, Boolean>() {
```

Comment:
(No space after "!")
Looking good, thank you for indulging. The example is now about as close as possible to the scala version